waku/v2/protocol/store/waku_store_protocol.go

```diff
@@ -323,20 +349,31 @@ func (store *WakuStore) Resume(ctx context.Context, pubsubTopic string, peerList
 		return -1, ErrNoPeersAvailable
 	}
 
-	messages, err := store.queryLoop(ctx, rpc, peerList)
+	queryLoopResults, err := store.queryLoop(ctx, rpc, peerList)
 	if err != nil {
 		store.log.Error("resuming history", zap.Error(err))
 		return -1, ErrFailedToResumeHistory
 	}
 
 	msgCount := 0
-	for _, msg := range messages {
-		if err = store.storeMessage(protocol.NewEnvelope(msg, store.timesource.Now().UnixNano(), pubsubTopic)); err == nil {
-			msgCount++
+	for _, r := range queryLoopResults {
+		if err == nil && r.response.GetError() != pb.HistoryResponse_NONE {
+			r.err = errors.New("invalid cursor")
+		}
+
+		if r.err != nil {
+			store.log.Warn("could not resume message history", zap.Error(r.err), logging.HostID("peer", r.peerID))
+			continue
+		}
+
+		for _, msg := range r.response.Messages {
+			if err = store.storeMessage(protocol.NewEnvelope(msg, store.timesource.Now().UnixNano(), pubsubTopic)); err == nil {
```

Comment:
Here duplicate messages are also counted as query is sent to multiple peers. 
Or does storeMessage return's error in case msg is already stored.

Reply:
It returns an error